contribute/release-guide.md

```diff
@@ -0,0 +1,500 @@
+
+Determine your Apache GPG Key and Key ID, as follows:
+
+    gpg --list-keys
+
+This will list your GPG keys. One of these should reflect your Apache account, for example:
+
+    --------------------------------------------------
+    pub   2048R/845E6689 2016-02-23
+    uid                  Nomen Nescio <anonymous@apache.org>
+    sub   2048R/BA4D50BE 2016-02-23
+
+Here, the key ID is the 8-digit hex string in the `pub` line: `845E6689`.
+
+Now, add your Apache GPG key to the Beam’s `KEYS` file both in [`dev`](https://dist.apache.org/repos/dist/dev/incubator/beam/KEYS) and [`release`](https://dist.apache.org/repos/dist/release/incubator/beam/KEYS) repositories at `dist.apache.org`. Follow the instructions listed at the top of these files.
+
+Configure `git` to use this key when signing code by giving it your key ID, as follows:
+
+    git config --global user.signingkey 845E6689
+
+You may drop the `--global` option if you’d prefer to use this key for the current repository only.
+
+#### Access to Apache Nexus repository
+
+Configure access to the [Apache Nexus repository](http://repository.apache.org/), which enables final deployment of releases to the Maven Central Repository.
+
+1. You log in with your Apache account.
+1. Confirm you have appropriate access by finding `org.apache.beam` under `Staging Profiles`.
+1. Navigate to your `Profile` (top right dropdown menu of the page).
+1. Choose `User Token` from the dropdown, then click `Access User Token`. Copy a snippet of the Maven XML configuration block.
+1. Insert this snippet twice into your global Maven `settings.xml` file, typically `${HOME}/.m2/settings.xml`. The end result should look like this, where `TOKEN_NAME` and `TOKEN_PASSWORD` are your secret tokens:
+
+        <settings>
+          <servers>
+            <server>
+              <id>apache.releases.https</id>
+              <username>TOKEN_NAME</username>
+              <password>TOKEN_PASSWORD</password>
+            </server>
+            <server>
+              <id>apache.snapshots.https</id>
+              <username>TOKEN_NAME</username>
+              <password>TOKEN_PASSWORD</password>
+            </server>
+          </servers>
+        </settings>
+
+#### Website development setup
+
+Get ready for updating the Beam website by following the [website development instructions](/contribute/contribution-guide/#website).
+
+### Create a new version in JIRA
+
+When contributors resolve an issue in JIRA, they are tagging it with a release that will contain their changes. With the release currently underway, new issues should be resolved against a subsequent future release. Therefore, you should create a release item for this subsequent release, as follows:
```

Comment:
Typically fixes will go into the development version, which is the next minor release. We additionally will cherry-pick from master into release branches, but only for hot fixes. What you are proposing is good, and a more flexible process, but we are just not there yet. For now I'll keep as-is but this is something we should consider doing in the near future.